frontend/src/speckleUtils.js

```diff
@@ -0,0 +1,84 @@
+import {
+  userInfoQuery,
+  streamListAllQuery,
+  streamSearchQuery,
+} from '@/speckleQueries'
+
+export const APP_NAME = process.env.VUE_APP_SPECKLE_APP_NAME
+export const SERVER_URL = process.env.VUE_APP_SPECKLE_SERVER_URL
+export const TOKEN = `${APP_NAME}.AuthToken`
+export const REFRESH_TOKEN = `${APP_NAME}.RefreshToken`
+export const CHALLENGE = `${APP_NAME}.Challenge`
+
+// Redirects to the Speckle server authentication page, using a randomly generated challenge. Challenge will be stored to compare with when exchanging the access code.
+export const goToSpeckleAuthPage = () => {
+  // Generate random challenge
+  const challenge =
+    Math.random().toString(36).substring(2, 15) +
+    Math.random().toString(36).substring(2, 15)
+
+  // Save challenge to localStorage
+  localStorage.setItem(CHALLENGE, challenge)
+  // Send user to auth page
+  // TODO: Why not constants?
+  window.location = `${process.env.VUE_APP_SPECKLE_SERVER_URL}/authn/verify/${process.env.VUE_APP_SPECKLE_APP_ID}/${challenge}`
```

Comment:
Is there a reason why `VUE_APP_SPECKLE_APP_ID` and `VUE_APP_SPECKLE_APP_SECRET` shouldn't be set as constants like the other environment variables?